typescript/common-resolvers/src/format-coco/coco-core/converters.ts

```diff
@@ -67,6 +71,36 @@ const convertLabelClassesToCocoCategories = (labelClasses: DbLabelClass[]) => {
   };
 };
 
+const convertGeometryToSegmentation = (
+  geometry: Geometry,
+  imageHeight: number
+): number[][] => {
+  const coordToSegmentationReducer = (
+    segmentation: number[],
+    [x, y]: number[],
+    _coordIndex: number,
+    _featureIndex: number,
+    _multiFeatureIndex: number,
+    geometryIndex: number
+  ) => {
+    // Only take the outer ring of the geometry
+    if (geometryIndex > 0) {
+      return [...segmentation];
+    }
+
+    return [...segmentation, x, imageHeight - y];
+  };
+
+  return geomReduce(
+    flatten(geometry),
+    (segmentations: number[][], currentGeometry) => [
+      ...segmentations,
+      coordReduce(currentGeometry, coordToSegmentationReducer, []),
+    ],
+    []
+  );
+};
+
```

Comment:
Since we know that we're manipulating only MultiPolygons, I was wondering whether we could avoid using `flatten`, `geomReduce`, `coordReduce` by just making a reduce in `geometry.coordinates`, and keeping the first linear ring for each polygon.

Reply:
I'd prefer to keep some geometry-agnostic code with `geomReduce` and others, it's a better practice I feel.

Reply:
I agree, still, is `flatten` strictly required l95 @ShauniArima ? Since I see `_multiFeatureIndex` l83 which makes me think that a multipolygon would be iterated on each enclosed polygon?